src/lib/bifrost.ts

```
@@ -127,3 +130,181 @@ function validateKeysetParams(threshold: number, totalMembers: number) {
+      }, timeout);
+
+      console.log(`[awaitShareEcho] Connecting node to relays...`);
+      await node.connect();
+      console.log(`[awaitShareEcho] Node connected. Listening for incoming echo.`);
+
+    } catch (error: any) {
+      console.error('[awaitShareEcho] General error during setup or connection:', error);
+      reject(error); 
+      cleanup(); 
+    }
+  });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)